client/src/main/kotlin/io/spine/examples/shareaware/client/Client.kt

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2023, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.client
+
+import com.google.protobuf.Message
+import io.grpc.ManagedChannelBuilder
+import io.spine.base.CommandMessage
+import io.spine.base.EntityState
+import io.spine.base.EventMessage
+import io.spine.client.Client
+import io.spine.client.ClientRequest
+import io.spine.client.EventFilter
+import io.spine.core.UserId
+import io.spine.examples.shareaware.WalletId
+import io.spine.examples.shareaware.wallet.command.CreateWallet
+import io.spine.examples.shareaware.wallet.event.WalletCreated
+import io.spine.util.Exceptions
+import java.util.*
+import java.util.concurrent.CompletableFuture
+
+/**
+ * Interacts with the app server via gRPC.
+ */
+public class DesktopClient private constructor(
+    host: String,
+    port: Int
+) {
+
+    public companion object {
+
+        @Volatile
+        private var instance: DesktopClient? = null
+
+        public fun init(host: String, port: Int): DesktopClient =
+            instance ?: synchronized(this) {
+                instance ?: DesktopClient(host, port).also { instance = it }
+            }
+    }
+
+    private val client: Client
+    private lateinit var user: UserId
+    private lateinit var walletId: WalletId
+
+    init {
+        val channel = ManagedChannelBuilder
+            .forAddress(host, port)
+            .usePlaintext()
+            .build()
+        client = Client
+            .usingChannel(channel)
+            .build()
+    }
+
+    /**
+     * Sends command to authenticate user in the system.
+     *
+     * A command `CreateWallet` is used for user creation because in the ShareAware
+     * there is no functional to register or authenticate the user, yet.
```

Comment:
"...because ShareAware server does not provide neither user registration nor authorisation features at the moment."